client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/HotRodAsyncCache.java

```diff
@@ -166,7 +166,7 @@ public CompletionStage<Void> clear(CacheOptions options) {
 
    @Override
    public <R> AsyncQuery<K, V, R> query(String query, CacheOptions options) {
-      return new HotRodAsyncQuery(query, options);
+      return new HotRodAsyncQuery<>();
```

Comment:
I would say just throw UnsupportedOperationException as well until we actually implement this.

Reply:
I kept the classes from the hotrod module but added UnsupportedOperationException to all the unimplemented methods.